pkg/idtools/idtools_test.go

```diff
@@ -273,3 +278,180 @@ func TestIsContiguous(t *testing.T) {
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1001,
+			expectedMode: 0o755 | os.ModeDir,
+		},
+		{
+			name:      "Invalid format: missing fields",
+			input:     []byte("1000:1001"),
+			expectErr: true,
+		},
+		{
+			name:      "Invalid UID format",
+			input:     []byte("invalid:1001:0644:file"),
+			expectErr: true,
+		},
+		{
+			name:         "valid symlink",
+			input:        []byte("1000:1001:0777:symlink"),
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1001,
+			expectedMode: 0o777 | os.ModeSymlink,
+		},
+		{
+			name:         "valid pipe",
+			input:        []byte("1000:1001:0666:pipe"),
+			expectErr:    false,
+			expectedUID:  1000,
+			expectedGID:  1001,
+			expectedMode: 0o666 | os.ModeNamedPipe,
+		},
+		{
+			name:          "valid block device",
+			input:         []byte("1000:1001:0600:block-5-6"),
+			expectErr:     false,
+			expectedUID:   1000,
+			expectedGID:   1001,
+			expectedMode:  0o600 | os.ModeDevice,
+			expectedMajor: 5,
+			expectedMinor: 6,
+		},
+		{
+			name:          "valid char device",
+			input:         []byte("1000:1001:0600:char-10-11"),
+			expectErr:     false,
+			expectedUID:   1000,
+			expectedGID:   1001,
+			expectedMode:  0o600 | os.ModeCharDevice | os.ModeDevice,
+			expectedMajor: 10,
+			expectedMinor: 11,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			stat, err := parseOverrideXattr(tt.input)
+			if tt.expectErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, stat)
+				assert.Equal(t, tt.expectedUID, stat.IDs.UID)
+				assert.Equal(t, tt.expectedGID, stat.IDs.GID)
+				assert.Equal(t, tt.expectedMode, stat.Mode)
+				assert.Equal(t, tt.expectedMajor, stat.Major)
+				assert.Equal(t, tt.expectedMinor, stat.Minor)
+			}
+		})
+	}
+}
+
+func TestFormatContainersOverrideXattrDevice(t *testing.T) {
+	tests := []struct {
+		name     string
+		uid      int
+		gid      int
+		mode     fs.FileMode
+		major    int
+		minor    int
+		expected string
+	}{
+		{"regular file", 1000, 1011, fs.FileMode(0o644), 0, 0, "1000:1011:0644:file"},
+		{"directory", 1001, 1012, fs.FileMode(0o755) | os.ModeDir, 0, 0, "1001:1012:0755:dir"},
+		{"symlink", 1002, 1013, fs.FileMode(0o777) | os.ModeSymlink, 0, 0, "1002:1013:0777:symlink"},
+		{"pipe", 1003, 1014, fs.FileMode(0o666) | os.ModeNamedPipe, 0, 0, "1003:1014:0666:pipe"},
+		{"socket", 1004, 1015, fs.FileMode(0o700) | os.ModeSocket, 0, 0, "1004:1015:0700:socket"},
+		{"block device", 1005, 1016, fs.FileMode(0o600) | os.ModeDevice, 8, 0, "1005:1016:0600:block-8-0"},
+		{"char device", 1006, 1017, fs.FileMode(0o600) | os.ModeDevice | os.ModeCharDevice, 1, 3, "1006:1017:0600:char-1-3"},
+
+		{"setuid bit", 1012, 1022, fs.FileMode(0o4755), 0, 0, "1012:1022:4755:file"},
+		{"setgid bit", 1013, 1023, fs.FileMode(0o2755), 0, 0, "1013:1023:2755:file"},
+		{"sticky bit", 1014, 1024, fs.FileMode(0o1777), 0, 0, "1014:1024:1777:file"},
```

Comment:
`07000` are not the right `fs.Mode*` values; AFAICS they have no assigned meaning.